### PR TITLE
Abstract component & navigation logic with Mixins

### PR DESCRIPTION
**Pull request by jcapona**
| Status  | Ticket/Issue |
| :---: | :--: |
| Ready/Hold | [Ticket](https://pi-top.atlassian.net/browse/<ticket-number>) |

#### Main changes
Abstract component behaviors/features into mixins:
- `Enterable` components define a `enterable_component` property that's used to determine the 'child' component that will get pushed into the stack.
- `Poppable` components receive a callback and it's used by itself when the page wants to be popped. This is used by project pages to pop themselves out from the stack once the project run finishes.
- `Actionable` components run actions using the `perform_action` method. Used by action pages and project pages; not really sure about this one.
- `HasGutterIcons` components can provide `top` and `bottom` icons for gutters.
- `Navigable` components define `go_next`, `go_previous` and `go_top` methods to navigate a list.
- `BlocksMiniscreenButtons` components define if the root app handles button presses based on their `block_buttons` property

#### Screenshots (feature, test output, profiling, dev tools etc)

[insert screenshots here]

#### Other notes (e.g. implementation quirks, edge cases, questions / issues)
-

#### Manual testing tips
-

#### Tag anyone who definitely needs to review or help
-

### CODE REVIEW

File: pt_miniscreen/core/components/selectable_list.py [hunk 1]
```diff
@@ -36,6 +36,14 @@ def selected_row(self):
     def is_empty(self):
         return len(self.rows) == 0
 
+    @property
+    def can_select_previous(self):
+        return 0 > self.state["selected_index"]
```
Comment (angusjfw) [inline pt_miniscreen/core/components/selectable_list.py:41]: this doesn't look right? 
```suggestion
        return self.state["selected_index"] > 0
```

File: pt_miniscreen/root.py [hunk 2]
```diff
@@ -152,90 +127,56 @@ def _wait_for_bootsplash_finish(self):
...
-
-    def select_next_row(self):
-        if self.can_select_row:
-            self.active_page.select_next_row()
-
-    def select_previous_row(self):
-        if self.can_select_row:
-            self.active_page.select_previous_row()
+    @property
+    def handles_button_events(self):
```
Comment (angusjfw) [inline pt_miniscreen/root.py:146]:
```suggestion
    def active_component_handles_button_events(self):
```
the fact that this was an @property, combined with it's name not mentioning the active component led to me being confused and thinking there was some important boolean as to whether the root is handling buttons or not.

**Review by olivierwilkinson — APPROVED** [review]
LGTM. I think it's fair to add some `return None` lines to the places CodeQL is complaining but up to you. 😄
